src/test/java/com/aws/greengrass/security/SecurityServiceTest.java

```diff
@@ -345,4 +353,103 @@ void GIVEN_key_and_cert_uri_WHEN_get_mqtt_builder_from_default_THEN_succeed() th
             assertThat(builder, IsNull.notNullValue());
         }
     }
+
+    @Test
+    void GIVEN_key_and_cert_uri_WHEN_getCertificateChain_and_providerImplemented_THEN_getCertChainFromProvider()
+            throws Exception {
+        URI keyUri = new URI("pkcs11:object=key-label");
+        URI certificateUri = new URI("file:///path/to/certificate");
+        when(mockKeyProvider.supportedKeyType()).thenReturn("PKCS11");
+        when(mockKeyProvider.getCertificateChain(keyUri, certificateUri)).thenReturn(Collections.emptyList());
+
+        service.registerCryptoKeyProvider(mockKeyProvider);
+        List<X509Certificate> certs = service.getCertificateChain(keyUri, certificateUri);
+        assertThat(certs, notNullValue());
+    }
+
+    @Test
+    void GIVEN_key_and_cert_uri_WHEN_get_cert_chain_and_provider_not_implemented_THEN_return_cert_chain()
+            throws Exception {
+        URI keyUri = new URI("pkcs11:object=key-label");
+        URI certificateUri = new URI("file:///path/to/certificate");
+        KeyManager[] mockKeyManagers = {mock(X509KeyManager.class)};
+        X509KeyManager x509KeyManager = (X509KeyManager) mockKeyManagers[0];
+        KeyPair mockKeyPair = mock(KeyPair.class);
+        String[] aliases = {"test"};
+        PrivateKey mockPrivateKey = mock(PrivateKey.class);
+        X509Certificate[] x509Certificates = new X509Certificate[1];
+        when(mockKeyProvider.getKeyManagers(keyUri, certificateUri)).thenReturn(mockKeyManagers);
+        when(mockKeyProvider.getKeyPair(keyUri, certificateUri)).thenReturn(mockKeyPair);
+        when(mockKeyPair.getPublic()).thenReturn(mock(PublicKey.class));
+        when(mockKeyPair.getPrivate()).thenReturn(mockPrivateKey);
+        when(mockKeyProvider.supportedKeyType()).thenReturn("PKCS11");
+        when(mockKeyProvider.getCertificateChain(keyUri, certificateUri))
+                .thenThrow(UnsupportedOperationException.class);
+        when(x509KeyManager.getClientAliases(any(), any())).thenReturn(aliases);
+        when(x509KeyManager.getPrivateKey(aliases[0])).thenReturn(mockPrivateKey);
+        when(x509KeyManager.getCertificateChain("test")).thenReturn(x509Certificates);
+
+        service.registerCryptoKeyProvider(mockKeyProvider);
+        List<X509Certificate> certs = service.getCertificateChain(keyUri, certificateUri);
+        assertThat(certs, notNullValue());
```

Comment:
Could we also assert that the value we get here is the cert that we expect. Just returning an List with a random X509Certificate or an empty array would make this test pass